docs/docs/mlflow-3/genai-agent.mdx

```diff
@@ -0,0 +1,201 @@
+---
+sidebar_position: 2
+---
+
+# GenAI Agent with MLflow 3.0
+
+This example demonstrates how to use MLflow to log and evaluate a generative AI agent.
+It showcases how to register prompts, log models, and assess their performance using evaluation datasets.
+The example also highlights the ability to track interactive traces and link them to the logged model for better observability.
+
+Prerequisite: ``pip install langchain langchain-community langchain-openai``
+
+## Create an agent
+
+First, we create an agent and utilize MLflow's prompt registry capability to manage and link prompts effectively.
+
+```python
+from langchain_core.prompts import ChatPromptTemplate
+from langchain_openai import ChatOpenAI
+
+import mlflow
+
+# register a prompt so we can link it when logging the model
+system_prompt = mlflow.register_prompt(
+    name="system_prompt",
+    template="You are a chatbot that can answer questions about IT.",
+    commit_message="Initial version of system prompt",
+)
+
+chat_model = ChatOpenAI(
+    model="gpt-4o-mini", temperature=0.1, max_completion_tokens=2000
+)
+chat_prompt = ChatPromptTemplate.from_messages(
+    [("system", system_prompt.to_single_brace_format()), ("user", "{messages}")]
+)
+chain = chat_prompt | chat_model
+print(chain.invoke({"messages": "What is MLflow?"}))
+```
+
+Switch to the **Prompts** tab to view the registered prompt and its details:
+
+<div className="center-div" style={{ width: "100%" }}>
+    ![The MLflow UI showing a prompt version](/images/mlflow-3/genai/system_prompt.png)
+</div>
+
+## Log the agent
+
+We recommend using a code-based logging approach for agents. Save your agent to a python file (e.g., `agent.py`)
```

Comment:
@serena-ruan Can we stop recommending code based logging and instead show `mlflow.create_external_model()` instead? This should be a small change.